install/migrations/update_10.0.x_to_10.1.0/recursive_groups.php

```diff
@@ -0,0 +1,62 @@
+<?php
+
+/**
+ * ---------------------------------------------------------------------
+ *
+ * GLPI - Gestionnaire Libre de Parc Informatique
+ *
+ * http://glpi-project.org
+ *
+ * @copyright 2015-2022 Teclib' and contributors.
+ * @copyright 2003-2014 by the INDEPNET Development Team.
+ * @licence   https://www.gnu.org/licenses/gpl-3.0.html
+ *
+ * ---------------------------------------------------------------------
+ *
+ * LICENSE
+ *
+ * This file is part of GLPI.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ * ---------------------------------------------------------------------
+ */
+
+/**
+ * @var DB $DB
+ * @var Migration $migration
+ */
+
+$migration->addConfig([
+    // Add new "recursive groups" configuration option
+    // If enabled, members of a given group will become implicits members of its
+    // children groups
+    // Disabled by default
+    'use_recursive_groups' => false,
+
+    // Store the last change to groups membership and/or configuration
+    // This allow to reset the group cache stored in the users session on demand
+    // Before that, users groups would not actualize until they were disconnected
+    'last_group_change' => $_SESSION['glpi_currenttime'],
+]);
+
+// Add new "recursive_membership" field on groups
+// It may be set to "No" to disable recursion for a given group
+// By default, it is enabled for all groups but wont be active unless the new
+// "use_recursive_groups" config option is also enabled
+if (!$DB->fieldExists('glpi_groups', 'recursive_membership')) {
+    $migration->addField('glpi_groups', 'recursive_membership', 'bool', [
+        'value' => true,
```

Comment:
Is this really necessary to have both a global config and an attribute in all groups ?

The `recursive_membership` could be set to false by default, and it would still be easy to activate the feature using massive actions.

Reply:
Yes, it was sold this way.
The config have a global switch to enable the feature, then you can disable it on specific group if needed.